testing/sdk_tasks.py

```diff
@@ -57,16 +57,21 @@ def fn():
         prefix_clause = ''
         if prefix:
             prefix_clause = ' starting with "{}"'.format(prefix)
-        log.info('Waiting for {} tasks{} to have updated ids:\n- Old tasks: {}\n- Current tasks: {}'.format(
-            len(old_task_ids), prefix_clause, sorted(old_task_ids), sorted(task_ids)))
-
-        all_updated = True
-        for id in task_ids:
-            if id in old_task_ids:
-                all_updated = False
-        if len(task_ids) < len(old_task_ids):
-            all_updated = False
-        return all_updated
+
+        old_set = set(old_task_ids)
+        new_set = set(task_ids)
+        newly_launched_set = (new_set - old_set)
+        all_updated = len(newly_launched_set) == len(new_set)
```

Comment:
This is validating that there are all new tasks, and that there are the same number?
Minor preference for something like

```python
task_overlap = new_set.intersection(old_set)
if len(new_set) == len(old_set) and len(task_overlap) == 0:
    return True
```

If that's the set of conditions that are being validated.

Reply:
1. Set operations via the set symbols should be clear.
  * new_set - old_set yields the difference, so newly_launched_set.
2. `newly_launched_set` should provide a clear intention of the result.

I believe (rather strongly) that such a name is much clearer than:
1. using a method call versus using the set symbols is possibly easier to read, but then we should also be using 2.plus(3) to yield 5.
2. `task_overlap` is an abstract name, so loses meaning as the value is used later in the method.

Reply:
I don't agree that the symbolic set operators are well-known.  set1 - set2 is clear enough.  Set1 & set2 is not, imo.

Regardless, this commentary was about making explicit the idea that we're checking for size equality and non-overlap, if that's what we're doing.  It wasn't about the name.

Reply:
@paegun the check for the lengths above is not sufficient to ensure that all tasks have been updated. We would also need to check that the size of the new set is the same as the old set. An empty new set, for example, would pass the check and return `all_updated`.

Reply:
True, I'll add a check to ensure that the old tasks are all gone.